src/Argu/Parsers/Common.fs

```diff
@@ -66,7 +66,23 @@ let postProcessResults (argInfo : UnionArgInfo) (ignoreMissingMandatory : bool)
             | Choice1Of2 ts, ts' when caseInfo.GatherAllSources.Value -> Array.append ts ts'
             | _, ts' -> ts'
 
+        let rec searchSub caseInfo =
```

Comment:
I'm sure this could be done in a less allocatey way, since you're only really looking for the first element

Reply:
Is that better? the tests pass. Is the wilcard return empty array the right call on line 78?

I'm not 100% sure I understand why unionArg.Cases.Value is a single item array. Is it because any duplicates will have been filtered out by the time this runs?

Reply:
Since the only `searchSub` callsite only checks for the first array element, it should really just return an optional.

Reply:
changed it to an option